File: src/main/java/spring/webmvc/domain/repository/ItemRepository.java
package spring.webmvc.domain.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import spring.webmvc.domain.model.entity.Item;

public interface ItemRepository {

	Page<Item> findAll(Pageable pageable, String name);

	Optional<Item> findById(Long id);

	Item save(Item item);

	List<Item> saveAll(Iterable<Item> items);

	void delete(Item item);
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/repository/ItemRepository.kt
package spring.web.kotlin.domain.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import spring.web.kotlin.domain.model.entity.Item

interface ItemRepository {
    fun findAll(pageable: Pageable, name: String?): Page<Item>
    fun findByIdOrNull(id: Long): Item?
    fun save(item: Item): Item
    fun saveAll(items: Iterable<Item>): List<Item>
    fun deleteById(id: Long)
}